Source/PeleLMEB.cpp

```diff
@@ -384,6 +385,139 @@ void PeleLM::getEBDistance(int a_lev,
+               }
+            }
+         });
+      }
+   }
+}
+
+void PeleLM::getEBDiff(int a_lev,
+                       const TimeStamp &a_time,
+                       MultiFab   &a_EBDiff,
+                       int        diffComp)
+{
+   // Get Geom / EB data
+   ProbParm const* lprobparm = prob_parm_d;
+   const auto geomdata = geom[a_lev].data();
+   const auto &ebfact = EBFactory(a_lev);
+   Array<const MultiCutFab*,AMREX_SPACEDIM> faceCentroid = ebfact.getFaceCent();
+
+   // Get diffusivity cell-centered
+   auto ldata_p = getLevelDataPtr(a_lev,a_time);
+
+   MFItInfo mfi_info;
+   if (Gpu::notInLaunchRegion()) mfi_info.EnableTiling().SetDynamic(true);
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+   for (MFIter mfi(a_EBDiff, mfi_info); mfi.isValid(); ++mfi)
+   {
+      const Box& bx = mfi.tilebox();
+      auto const& flagfab = ebfact.getMultiEBCellFlagFab()[mfi];
+      auto const& flag    = flagfab.const_array();
+      const auto& ebdiff  = a_EBDiff.array(mfi);
+      const auto& diff_cc = ldata_p->diff_cc.const_array(mfi, diffComp);
+
+      if ( flagfab.getType(bx) == FabType::covered ) {              // Set to zero
+         AMREX_PARALLEL_FOR_3D(bx, i, j, k, {ebdiff(i,j,k) = 0.0;});
+      } else if ( flagfab.getType(bx) == FabType::regular ) {       // Set to zero
+         AMREX_PARALLEL_FOR_3D(bx, i, j, k, {ebdiff(i,j,k) = 0.0;});
+      } else {
+         AMREX_D_TERM( const auto& ebfc_x = faceCentroid[0]->array(mfi);,
+                       const auto& ebfc_y = faceCentroid[1]->array(mfi);,
+                       const auto& ebfc_z = faceCentroid[2]->array(mfi););
+         amrex::ParallelFor(bx, [=]
+         AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+         {
+            // Regular/covered cells -> 0.0
+            if ( flag(i,j,k).isCovered() ||
+                 flag(i,j,k).isRegular()) {
+               ebdiff(i,j,k) = 0.0;
+            } else { // cut-cells
+               // Get the EBface centroid coordinates
+               const amrex::Real* dx = geomdata.CellSize();
+               const amrex::Real* prob_lo = geomdata.ProbLo();
+               const amrex::Real xcell[AMREX_SPACEDIM] = {AMREX_D_DECL(prob_lo[0] + (i + 0.5) * dx[0],
+                                                                       prob_lo[1] + (j + 0.5) * dx[1],
+                                                                       prob_lo[2] + (k + 0.5) * dx[2])};
+               const amrex::Real xface[AMREX_SPACEDIM] = {AMREX_D_DECL(xcell[0] + ebfc_x(i,j,k) * dx[0],
+                                                                       xcell[1] + ebfc_y(i,j,k) * dx[1],
+                                                                       xcell[2] + ebfc_z(i,j,k) * dx[2])};
+               amrex::Real ebflagtype = 0.0;
+               setEBType(xface, ebflagtype, geomdata, *lprobparm);
+               ebdiff(i,j,k) = diff_cc(i,j,k) * ebflagtype;
```

Comment:
Just a note that if running with an LES model this will only get the molecular component of the thermal diffusivity because the turbulent component is added when moving from cc to fc. I believe this behavior is the same for how the no slip condition on EBs is presently implemented, and is fine. If using WALE and enough resolution at the boundaries the turbulent part should go to 0 anyway, and if not then we'd want to add a wall model.